src/Features/Core/Portable/UpgradeProject/AbstractUpgradeProjectCodeFixProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeAnalysis.UpgradeProject
+{
+    internal abstract partial class AbstractUpgradeProjectCodeFixProvider : CodeFixProvider
```

Comment:
It's not clear to me if this abstract type here is holding it's own weight, since it's just forwarding a method to a different method.